parser/ast/context.go

```diff
@@ -0,0 +1,43 @@
+package ast
+
+import (
+	"context"
+	"sync"
+)
+
+type modulesKey struct{}
+
+func WithModules(ctx context.Context, mods *ModuleLookup) context.Context {
+	return context.WithValue(ctx, modulesKey{}, mods)
+}
+
+func Modules(ctx context.Context) *ModuleLookup {
+	mods, ok := ctx.Value(modulesKey{}).(*ModuleLookup)
+	if !ok {
+		return NewModules()
+	}
+	return mods
+}
+
+type ModuleLookup struct {
+	mods map[string]*Module
+	mu   sync.Mutex
+}
+
+func NewModules() *ModuleLookup {
+	return &ModuleLookup{
+		mods: make(map[string]*Module),
+	}
+}
+
+func (ml *ModuleLookup) Get(filename string) *Module {
```

Comment:
This isn't used anywhere, right?

Reply:
It's not used atm, I just wanted the eventual review of the debugger to be easier.